src/Microsoft.ML.Data/Training/TrainerUtils.cs

```diff
@@ -382,6 +383,24 @@ public static SchemaShape.Column MakeR4ScalarWeightColumn(string weightColumn)
                 return null;
             return new SchemaShape.Column(weightColumn, SchemaShape.Column.VectorKind.Scalar, NumberType.R4, false);
         }
+
+        /// <summary>
+        /// Check that the label, feature, weights is not supplied in the args the constructor.
+        /// Those parameters should be internal if they are not used from the maml help code path.
+        /// </summary>
+        public static void CheckArgumesDefaultColNames(string defaultColName, string argValue)
+        {
+            if (argValue != defaultColName)
+                throw Contracts.Except($"Don't supply a value for the {defaultColName} column in the arguments, as it will be ignored. Specify them in the loader, or constructor instead instead.");
```

Comment:
Everywhere you call this you have host or environment. Why you don't want to use it?